src/__tests__/testFunctions.ts

```diff
@@ -39,6 +39,39 @@ export function getTestData(name: string) {
     return cachedTestData[name] || (cachedTestData[name] = readFileSync(`testdata/${name}/data.bin`).toString());
 }
 
+// eslint-disable-next-line  @typescript-eslint/no-explicit-any
+function runGeneralTests(identifier: string, compressFunc: (input: any) => any, decompressFunc: (input: any) => any) {
```

Comment:
Typedocs to be added, and use generics rather than `any` - but abstraction is good :-)

Reply:
I'm sure it's possible, but might be worth merging this in and fixing later